core/ai.go

```diff
@@ -41,6 +42,28 @@ func (s JSONRat) String() string {
 	return s.FloatString(2)
 }
 
+func PipelineToCapability(pipeline string) Capability {
+	if len(pipeline) == 0 {
+		return Capability_Unused
+	}
+	runes := []rune(pipeline)
+	runes[0] = unicode.ToUpper(runes[0])
+	for i, r := range runes {
+		if r == '-' {
+			runes[i] = ' '
+		}
+	}
+	pipelineName := string(runes)
+	for cap, desc := range CapabilityNameLookup {
+		if pipelineName == desc {
+			return cap
+		}
+	}
+
+	//no capability description matches name
+	return Capability_Unused
```

Comment:
I wonder if this should be an explicit error instead (or the caller check this `Unused` value) to avoid silent bad configurations. Otherwise one might be missing a configuration that they swear they are making on the ir JSON config file which could have bad implications (e.g. paying more than they were willing to). In general I think it's better to fail explicitly for bad configs than to ignore them.

Reply:
I agree that it would be better to update this to return `(Capability, error)` to make it more obvious it should be checked by the caller.

Updated and added a test for this function in ai_test.go